src/CarbonAware.WebApi/test/integrationTests/CarbonAwareControllerTests.cs

```diff
@@ -48,16 +49,120 @@ public async Task BestLocations_ReturnsOK(DateTimeOffset start, DateTimeOffset e
         _dataSourceMocker.SetupDataMock(start, end, location);
 
         //Call the private method to construct with parameters
-        var endpointURI = ConstructDateQueryURI(bestLocationsURI, location, start, end);
+        var queryStrings = new Dictionary<string, string>();
+        queryStrings["location"] = location;
+        queryStrings["time"] = $"{start:O}";
+        queryStrings["toTime"] = $"{end:O}";
+
+        var endpointURI = ConstructUriWithQueryString(bestLocationsURI,queryStrings);
 
         //Get response and response content
         var result = await _client.GetAsync(endpointURI);
 
         Assert.That(result, Is.Not.Null);
         Assert.That(result.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+    }
+
+    [TestCase("location", "", TestName = "empty location query string")]
+    [TestCase("non-location-param", "", TestName = "location param not present")]
+    public async Task BestLocations_EmptyLocationQueryString_ReturnsBadRequest(string queryString, string value)
+    {
+        //Call the private method to construct with parameters
+        var queryStrings = new Dictionary<string, string>();
+        queryStrings[queryString] = value;
+
+        var endpointURI = ConstructUriWithQueryString(bestLocationsURI, queryStrings);
+
+        //Get response and response content
+        var result = await _client.GetAsync(endpointURI);
+
+        Assert.That(result, Is.Not.Null);
+        Assert.That(result.StatusCode, Is.EqualTo(HttpStatusCode.BadRequest));
+    }
+
+    [Test]
+    public async Task EmissionsForecastsCurrent_JSON_ReturnsNotImplemented()
```

Comment:
Not sure how I feel about splitting the test like this into JSON and WattTime when the test does the same thing except one assert is different. One thought I had was make it one test but have the list of "unimplemented datasources" and before the final assert, check if the `_datasource` is in the unimplemented category. If it is, assert on the `not implemented` if it's not, assert on the `ok`.

Not sure if that's _that much better_ than the current implementation so 🤷🏽‍♀️ take with grain of salt